datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1229,11 +1229,19 @@ impl SMJStream {
             }
             buffered_batch.null_joined.clear();
 
-            // For buffered rows which are joined with streamed side but doesn't satisfy the join filter
+            // For buffered row which is joined with streamed side rows but all joined rows
+            // don't satisfy the join filter
             if output_not_matched_filter {
+                let buffered_indices = buffered_batch
```

Comment:
perhaps we can name it `not_matched_buffered_indices` to let know what exactly those indices are

Reply:
Okay.